lib/rubocop/cop/rspec/cop.rb

```diff
@@ -3,72 +3,8 @@
 module RuboCop
   module Cop
     module RSpec
-      # @abstract parent class to RSpec cops
-      #
-      # The criteria for whether rubocop-rspec analyzes a certain ruby file
-      # is configured via `AllCops/RSpec`. For example, if you want to
-      # customize your project to scan all files within a `test/` directory
-      # then you could add this to your configuration:
-      #
-      # @example configuring analyzed paths
-      #   # .rubocop.yml
-      #   # AllCops:
-      #   #   RSpec:
-      #   #     Patterns:
-      #   #     - '_test.rb$'
-      #   #     - '(?:^|/)test/'
-      class Cop < ::RuboCop::Cop::Base
-        include RuboCop::RSpec::Language
-        include RuboCop::RSpec::Language::NodePattern
-
-        DEFAULT_CONFIGURATION =
-          RuboCop::RSpec::CONFIG.fetch('AllCops').fetch('RSpec')
-
-        DEFAULT_PATTERN_RE = Regexp.union(
-          DEFAULT_CONFIGURATION.fetch('Patterns')
-                               .map(&Regexp.public_method(:new))
-        )
-
-        # Invoke the original inherited hook so our cops are recognized
-        def self.inherited(subclass)
-          RuboCop::Cop::Cop.inherited(subclass)
-        end
-
-        def relevant_file?(file)
-          relevant_rubocop_rspec_file?(file) && super
-        end
-
-        private
-
-        def relevant_rubocop_rspec_file?(file)
-          rspec_pattern =~ file
-        end
-
-        def rspec_pattern
-          if rspec_pattern_config?
-            Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
-          else
-            DEFAULT_PATTERN_RE
-          end
-        end
-
-        def all_cops_config
-          config
-            .for_all_cops
-        end
-
-        def rspec_pattern_config?
-          return unless all_cops_config.key?('RSpec')
-
-          all_cops_config.fetch('RSpec').key?('Patterns')
-        end
-
-        def rspec_pattern_config
-          all_cops_config
-            .fetch('RSpec', DEFAULT_CONFIGURATION)
-            .fetch('Patterns')
-        end
-      end
+      # @deprecated Use ::RuboCop::Cop::RSpec::Base instead
+      Cop = Base
```

Comment:
should it produce some deprecation warning?

Reply:
I am not sure how to do that. An initializer hook? E.g.
```rb
def initialize(*)
  warn '`::RuboCop::Cop::RSpec::Cop` is deprecate. Please use `::RuboCop::Cop::RSpec::Base` instead.'
  super
end
```

Reply:
I'm not sure if it's really necessary.

If you feel it is, the following will do (just in case not to rely on `super`):
```ruby
class Cop < Base
  def self.inherited(base)
    warn '`::RuboCop::Cop::RSpec::Cop` is deprecated. Please use `::RuboCop::Cop::RSpec::Base` instead.'
  end
end
```

Reply:
I was thinking the other day what is the best way to track things that we'd like to drop/improve in a major version when RuboCop is 2.0, a separate ticket, a project, a `2.0` tag, a code annotation.

Anything would do I suppose. Do you have any preference? 

I only have this deprecation and those two pulls in mind https://github.com/rubocop-hq/rubocop-rspec/pull/940 (cop retired). Is there anything I'm missing?

Reply:
I think a separate ticket and a `2.0` milestone (to tag issues/PRs) would do.

Reply:
Maybe it's time for rubocop-rubocop, containing cops on how to properly write cops, that could be used by extension developers :) Internal investigations brought to the next level ;)

Reply:
This reminds me of a [style-guide-style-guide](https://github.com/pirj/meta-style-guide) :D